File: Easy/cleanUp/cleanUp.java
import java.util.Scanner;
import java.util.Arrays;

public class cleanUp{
	public static void main(String[] args) {
		int T;
		Scanner scan = new Scanner(System.in);

		while(true){
			T = scan.nextInt();

			if(T<=0 || T>50){
				System.out.println("T must be 1 <= T <= 50");
			}else{
				break;
			}
		}

		int n;
		int m;

		while(T>0){

			while(true){
				n = scan.nextInt();
				m = scan.nextInt();

				if(m<0 || m>n){
					System.out.println("m must be 0 <= m <= n");
				}else if(n<m || n>1000){
					System.out.println("n must be m <= n <= 1000");
				}else{
					break;
				}
			}

			int[] done = new int[m];

			for(int i=0;i<m;i++){
				while(true){
					done[i] = scan.nextInt();

					if(done[i]<1 || done[i]>n){
						System.out.println("done["+i+"] must be 1 <= done[i] <= n");
					}else{
						break;
					}
				}
			}

			boolean[] arr = new boolean[n];

			Arrays.sort(done);

			for(int i=0;i<m;i++){
				done[i]--;
			}

			for(int i=0;i<m;i++){
				for(int j=0;j<n;j++){
					if(done[i]==j){
						arr[j]=true;
					}
				}
			}

			int count = 0;

			for(int i=0;i<n;i++){
				if(arr[i]==false){
					count++;
					if(count%2!=0){
						System.out.print(i+1+ " ");
					}
				}
			}

			System.out.println();
			count = 0;

			for(int i=0;i<n;i++){
				if(arr[i]==false){
					count++;
					if(count%2==0){
						System.out.print(i+1+ " ");
					}
				}
			}

			System.out.println();

			T--;
		}
	}
}

File: Easy/cleanUp/cleanUp.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

int compare(const void *a, const void *b){
	return *(int*)a-*(int*)b;
}

int main(void){
	int T;

	while(1){
		scanf("%d",&T);

		if(T<=0 || T>50){
			printf("T must be 1 <= T <= 50\n");
		}else{
			break;
		}	
	}

	int n;
	int m;

	while(T>0){

		while(1){
			scanf("%d %d",&n,&m);

			if(m<0 || m>n){
				printf("m must be 0 <= m <= n\n");
			}else if(n<m || n>1000){
				printf("n must be m <= n <= 1000\n");
			}else{
				break;
			}	
		}

		int done[m];

		for(int i=0;i<m;i++){
			while(1){
				scanf("%d",&done[i]);

				if(done[i]<1 || done[i]>n){
					printf("done[%d] must be 1 <= done[i] <= n\n",i);
				}else{
					break;
				}
			}
		}

		bool arr[n];

		for(int i=0;i<n;i++){
			arr[i] = true;
		}

		qsort(done,m,sizeof(int),compare);

		for(int i=0;i<m;i++){
			done[i]--;
		}

		for(int i=0;i<m;i++){
			for(int j=0;j<n;j++){
				if(done[i]==j){
					arr[j]=false;
				}
			}
		}

		int count = 0;

		for(int i=0;i<n;i++){
			if(arr[i]==true){
				count++;
				if(count%2!=0){
					printf("%d ",i+1);
				}
			}
		}

		printf("\n");

		count = 0;

		for(int i=0;i<n;i++){
			if(arr[i]==true){
				count++;
				if(count%2==0){
					printf("%d ",i+1);
				}
			}
		}

		printf("\n");

		T--;
	}
	
	return 0;
}